HomeworkBasic1.ipynb

```diff
@@ -0,0 +1,192 @@
+     "text": [
+      "45\n"
+     ]
+    }
+   ],
+   "source": [
+    "a = [0,1,2,3,4,5,6,7,8,9]\n",
+    "b = sum(a)\n",
+    "print (b)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Assignment 3\n",
+    "\n",
+    "Adding to assignment 2, implement the same functionality but allow the array to be a mix of numbers and numbers represented as strings, e.g. [1, 5, '7', 5, '10']."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "224\n"
+     ]
+    }
+   ],
+   "source": [
+    "num1 = 100\n",
+    "num2 = 10\n",
+    "num3 = 50\n",
+    "a = [num1, 5, num2, 10, num3, 49]\n",
+    "b = sum(a)\n",
+    "print(b)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Assignment 4\n",
+    "\n",
+    "Let's build a small database program. The goal is to let the user build a database of pets. The database is a dictionary, in which a pet's name acts as the key and the value holds the type of animal (e.g. 'kitty': 'cat'). We'll assume that no two animals have the same name. Use a `while` loop to allow a user to add entries by first asking for the name of the pet and then for its type. Then ask the user if they want to add another entry.\n",
+    "    \n",
+    "*Tip*: the while loop condition can be a statement such as variables equals or does not equal a value."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "dict_keys(['peter', 'MM', 'kitty', 'yu'])\n",
+      "dict_values(['horse', 'dog', 'cat', 'fish'])\n",
+      "peter horse\n",
+      "MM dog\n",
+      "kitty cat\n",
+      "yu fish\n"
+     ]
+    }
+   ],
+   "source": [
+    "dict1 = {\"peter\": \"horse\", \"MM\": \"dog\", \"kitty\": \"cat\", \"yu\": \"fish\"}\n",
+    "print(dict1.keys())\n",
+    "print(dict1.values())\n",
+    "for k, v in dict1.items():\n",
+    "    print(k, v)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Assignment 5\n",
+    "\n",
+    "Split a list of unknown length into 4 equal sized parts. You can make the assumption that the length of the list is a multiple of 4. Print the resulting lists.\n",
+    "\n",
+    "*Tip*: A list can contain multiple lists."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 46,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[[4, 8, 12, 16], [20, 24, 28, 32], [36, 40, 44, 48], [52, 56, 60, 64]]\n"
+     ]
+    }
+   ],
+   "source": [
+    "list = [4, 8, 12, 16, 20, 24, 28, 32, 36, 40, 44, 48, 52, 56, 60, 64]\n",
+    "x = [list[i*4:(i+1)*4] \n",
+    "for i in range((len(list)+4-1) //4 )]\n",
+    "print (x) "
```

Comment:
are you sure you understand what you've implemented here? besides that this is pretty hard to read, you also used "list" as a variable name which overrides the function `list()`, which is something you rarely want to do.